x-pack/plugins/index_management/server/routes/api/templates/register_create_route.ts

```diff
@@ -15,54 +15,44 @@ import { saveTemplate, doesTemplateExist } from './lib';
 
 const bodySchema = templateSchema;
 
-export function registerCreateRoute({ router, lib }: RouteDependencies) {
+export function registerCreateRoute({ router, lib: { handleEsError } }: RouteDependencies) {
   router.post(
     { path: addBasePath('/index_templates'), validate: { body: bodySchema } },
-    async (ctx, req, res) => {
-      const { callAsCurrentUser } = ctx.dataManagement!.client;
-      const template = req.body as TemplateDeserialized;
-      const {
-        _kbnMeta: { isLegacy },
-      } = template;
-
-      // Check that template with the same name doesn't already exist
-      const templateExists = await doesTemplateExist({
-        name: template.name,
-        callAsCurrentUser,
-        isLegacy,
-      });
-
-      if (templateExists) {
-        return res.conflict({
-          body: new Error(
-            i18n.translate('xpack.idxMgmt.createRoute.duplicateTemplateIdErrorMessage', {
-              defaultMessage: "There is already a template with name '{name}'.",
-              values: {
-                name: template.name,
-              },
-            })
-          ),
-        });
-      }
+    async (context, request, response) => {
+      const { client } = context.core.elasticsearch;
+      const template = request.body as TemplateDeserialized;
 
       try {
-        // Otherwise create new index template
-        const response = await saveTemplate({ template, callAsCurrentUser, isLegacy });
+        const {
+          _kbnMeta: { isLegacy },
+        } = template;
+
+        // Check that template with the same name doesn't already exist
+        const { body: templateExists } = await doesTemplateExist({
```

Comment:
I moved all of this logic inside the `try/catch` because I figured we'd want to catch ES errors thrown by these helpers.